src/leveled_bookie.erl

```diff
@@ -1849,22 +1886,43 @@ set_options(Opts, Monitor) ->
         }.
 
 
--spec return_snapfun(book_state(), store|ledger, 
-                        tuple()|no_lookup|undefined, 
-                        boolean(), boolean()) 
-                            -> fun(() -> {ok, pid(), pid()|null}).
+-spec return_snapfun(
+    book_state(), store|ledger,
+    tuple()|no_lookup|undefined,
+    boolean(), boolean()) 
+        -> fun(() -> {ok, pid(), pid()|null, fun(() -> ok)}).
 %% @doc
 %% Generates a function from which a snapshot can be created.  The primary
 %% factor here is the SnapPreFold boolean.  If this is true then the snapshot
 %% will be taken before the Fold function is returned.  If SnapPreFold is
 %% false then the snapshot will be taken when the Fold function is called.
 %%
 %% SnapPrefold is to be used when the intention is to queue the fold, and so
-%% claling of the fold may be delayed, but it is still desired that the fold
+%% calling of the fold may be delayed, but it is still desired that the fold
 %% represent the point in time that the query was requested.
+%% 
+%% Also returns a function which will close any snapshots to be used in the
+%% runners post-query cleanup action
+%% 
+%% When the bookie is a snapshot, a fresh snapshot should not be taken, the
+%% previous snapshot should be used instead.  Also the snapshot should not be
+%% closed as part of the post-query activity as the snapshot may be reused, and
+%% should be manually closed.
 return_snapfun(State, SnapType, Query, LongRunning, SnapPreFold) ->
-    case SnapPreFold of
-        true ->
+    CloseFun =
```

Comment:
Sanity check:
Following flow it can be seen that `pcl_close` always returns `ok` if it returns... which means that `ink_close` will be called eventually. But also ink_close will always return `ok` if it returns (there is no corresponding match in line 1918)

Alternative is to check for Pid right away when function is created:
```
fun(LS0, JS0) when is_pid(JS0) ->
    fun() -> ok = leveled_penciller:pcl_close(LS0), leveled_inker:ink_close(JS0) end;
    (LS0, _) ->
       fun() -> leveled_penciller:pcl_close(LS0) end
   end
```

Reply:
I think I'm happy with it as-is.  Double-checked as well `ok` is the only return.